Error: The terraform-provider-azurerm_v4.1.0_x5 plugin crashed!

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

4.241.26

### AzureRM Provider Version

4.1.0

### Affected Resource(s)/Data Source(s)

azurerm_monitor_metric_alert, cosmos etc

### Terraform Configuration Files

```hcl
resource "azurerm_monitor_metric_alert" "app_service_health_check_alert" {
  resource_group_name = "example"
  for_each            = local.health_check_alerts_at_locations

  name        = "${azurerm_windows_web_app.app[each.value.location].name}-health-check-${each.value.severity_name}"
  scopes      = [azurerm_windows_web_app.app[each.value.location].id]
  description = "Alarm triggered by average health check below ${each.value.threshold} on service."
  tags        = var.tags

  frequency     = "PT1M"
  window_size   = each.value.window_size
  severity      = each.value.severity_code
  auto_mitigate = true

  criteria {
    metric_namespace = "Microsoft.Web/sites"
    metric_name      = "HealthCheckStatus"
    aggregation      = "Average"
    operator         = "LessThan"
    threshold        = each.value.threshold
  }

  action {
    action_group_id = ""
  }
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v4.1.0_x5 plugin:

panic: interface conversion: interface {} is []interface {}, not string

goroutine 1024 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/cosmos/migration.(*CosmosDBAccountV0toV1).UpgradeFunc.CosmosDBAccountV0toV1.UpgradeFunc.func1({0x0?, 0x0?}, 0x0?, {0x0?, 0x0?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/cosmos/migration/cosmosdb_account.go:560 +0xf0
github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk.StateUpgrades.func1({0x8b0c578, 0xc0030415c0}, 0x0?, {0x729e040, 0xc001adb680})
	github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk/state_upgrades.go:55 +0x5e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).upgradeJSONState(0xc000971c20, {0x8b0c578, 0xc0030415c0}, 0xd?, 0x832002a?, 0x14?)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:491 +0x75
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).UpgradeResourceState(0xc000971c20, {0x8b0c578?, 0xc0030414d0?}, 0xc002e33240)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:367 +0x3fa
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).UpgradeResourceState(0x8b0c5b0?, {0x8b0c578?, 0xc0030411d0?}, 0xc002e33240)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_UpgradeResourceState.go:36 +0x1a2
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).UpgradeResourceState(0xc000880460, {0x8b0c578?, 0xc0030400f0?}, 0xc001aca960)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:750 +0x27b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_UpgradeResourceState_Handler({0x7fbde60?, 0xc000880460}, {0x8b0c578, 0xc0030400f0}, 0xc003d9c480, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:446 +0x16c
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001cd000, {0x8b0c578, 0xc003040060}, {0x8b38f60, 0xc000250480}, 0xc001405320, 0xc0013a85a0, 0xe173738, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc0001cd000, {0x8b38f60, 0xc000250480}, 0xc001405320)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 16
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x135

Error: The terraform-provider-azurerm_v4.1.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

Just trying to delete these  resources from azure and plan failed with above error. expected behaviour is plan showing resources to destroy

### Actual Behaviour

plan failed

### Steps to Reproduce

terraform plan

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/cosmos/migration/cosmosdb_account.go
+++ b/internal/services/cosmos/migration/cosmosdb_account.go
@@ -557,12 +557,13 @@ func (c CosmosDBAccountV0toV1) Schema() map[string]*pluginsdk.Schema {
 
 func (c CosmosDBAccountV0toV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
 	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
-		ipfString := rawState["ip_range_filter"].(string)
-		ipfSet := make([]string, 0)
-		if ipfString != "" {
-			ipfSet = strings.Split(ipfString, ",")
+		if ipfString, ok := rawState["ip_range_filter"].(string); ok {
+			ipfSet := make([]string, 0)
+			if ipfString != "" {
+				ipfSet = strings.Split(ipfString, ",")
+			}
+			rawState["ip_range_filter"] = ipfSet
 		}
-		rawState["ip_range_filter"] = ipfSet
 
 		return rawState, nil
 	}

